Kernel/Storage/StorageManagement.cpp

```diff
@@ -105,7 +105,8 @@ UNMAP_AFTER_INIT void StorageManagement::enumerate_pci_controllers(bool force_pi
             auto subclass_code = static_cast<SubclassID>(device_identifier.subclass_code().value());
 #if ARCH(I386) || ARCH(X86_64)
             if (subclass_code == SubclassID::IDEController && kernel_command_line().is_ide_enabled()) {
-                m_controllers.append(PCIIDELegacyModeController::initialize(device_identifier, force_pio));
+                auto ide_controller = MUST(PCIIDELegacyModeController::initialize(device_identifier, force_pio));
```

Comment:
```suggestion
                auto ide_controller = PCIIDELegacyModeController::initialize(device_identifier, force_pio).release_value_but_fixme_should_propagate_errors();
```
?

Reply:
Hmm, right, forgot about this :)

Reply:
Oh - I didn't forget about this... I use MUST() here because currently initialization should never fail in the StorageManagement code, because of how this happens during boot. When we support PCI hotplug, it will be valuable to do something else, but this is the *correct* behavior for now.

Reply:
Hmm, i still don't think its appropriate to use `MUST` in that case. You could check for an error, print a message to `dmesgln` saying something like: "Cannot continue boot, because ....", and then panic. To my understanding, using `MUST` at the call-site means that the caller knows that the functions will not return an error, but in this case the hardware could still return an error for any reason.

Reply:
That's right, so I changed this to something else which seems far better now :)